ff rework

MTLehner opened the pull request:
## Description

Rework of the forcefield structure in PyGromos

Introducing a generic_forcfield and all other forcefield have this class as super. This makes code more readable and unifies the structure.

SalomeRonja commented:
Hey @MTLehner 
I just added amber2gromos to the reworked ff structure, and updated the corresponding example notebook. I really like the changes you made, it's a lot cleaner now than before! :)

MTLehner commented:
> Hey @MTLehner I just added amber2gromos to the reworked ff structure, and updated the corresponding example notebook. I really like the changes you made, it's a lot cleaner now than before! :)

Thank you! Also very nice from your side. Thank you for the quick rework

pultar reviewed (approved):
My local tests are still working! Good job! 😎
